tutorials/persistance-comparison-series/PlayerPrefs/player_prefs.md

```diff
@@ -0,0 +1,436 @@
+
+```cs
+private void Update() // 6
+{
+    // Check if a key was pressed.
+    if (Input.GetKey(KeyCode.LeftShift)) // 7
+    {
+        // Set the LeftShift key.
+        keyPressed = KeyCode.LeftShift; // 8
+    }
+    else if (Input.GetKey(KeyCode.LeftControl)) // 7
+    {
+        // Set the LeftControl key.
+        keyPressed = KeyCode.LeftControl; // 8
+    }
+    else // 9
+    {
+        // In any other case reset to default and consider it unmodified.
+        keyPressed = default; // 10
+    }
+}
+```
+
+The same triplet can then also be found in the click detection:
+
+```cs
+private void OnMouseDown()
+{
+    // Check if a key was pressed.
+    if (keyPressed == KeyCode.LeftShift) // 11
+    {
+        // Increment the hit count and set it to PlayerPrefs.
+        hitCountShift++; // 12
+        PlayerPrefs.SetInt(HitCountKeyShift, hitCountShift); // 15
+    }
+    else if (keyPressed == KeyCode.LeftControl) // 11
+    {
+        // Increment the hit count and set it to PlayerPrefs.
+        hitCountControl++; // 12
+        PlayerPrefs.SetInt(HitCountKeyControl, hitCountControl); // 15
+    }
+    else // 13
+    {
+        // Increment the hit count and set it to PlayerPrefs.
+        hitCountUnmodified++; // 14
+        PlayerPrefs.SetInt(HitCountKeyUnmodified, hitCountUnmodified); // 15
+    }
+
+    // Persist the data to disk.
+    PlayerPrefs.Save(); // 16
+}
+```
+
+First we check if one of those two was held down while the click happened (11) and if so, increment the corresponding hit counter (12). If not (13), the `unmodfied` counter has to be incremented (14).
+
+Finally, we need to set each of those three counters individually (15) via `PlayerPrefs.SetInt()` using the three keys we defined earlier.
+
+Like in the previous example, we also call `Save()` (16) at the end to make sure, data does not get lost if the game does not end normally.
+
+When switching back to the Unity editor, the script on the capsule should now look like this:
+
+<img src="images/05_extended_version.jpg" alt="Extended example" width="500"/>
+
+## More complex data
+
+(see `PlayerPrefsExampleJson.cs` in the repository for the finished version)
+
+In the previous two sections we have seen how to handle two simple examples of persisting data in `PlayerPrefs`. What if they get more complex than that? What if you want to structure and group data together?
+
+One possible approach would be to use the fact that `PlayerPrefs` can hold a `string` and save a `JSON` in there.
+
+First we need to figure out how to actually transform our data into JSON. The .NET framework as well as the `UnityEngine` framework offer a JSON serialize and deserializer to do this job for us. Both behave very similar but we will use Unity's own [`JsonUtility`](https://docs.unity3d.com/ScriptReference/JsonUtility.html) which [performs better in Unity than other similar JSON solution](https://docs.unity3d.com/Manual/JSONSerialization.html).
```

Comment:
```suggestion
First we need to figure out how to actually transform our data into JSON. The .NET framework as well as the `UnityEngine` framework offer a JSON serializer and deserializer to do this job for us. Both behave very similar but we will use Unity's own [`JsonUtility`](https://docs.unity3d.com/ScriptReference/JsonUtility.html) which [performs better in Unity than other similar JSON solutions](https://docs.unity3d.com/Manual/JSONSerialization.html).
```

Reply:
As a reminder, this is marked as resolved but the changes aren't present.